.github/workflows/build-and-upload-staging.yml

```diff
@@ -66,8 +66,7 @@ jobs:
     uses: ./.github/workflows/appsignals-e2e-test.yml
     secrets: inherit
     concurrency:
-      group: 'appsignals-cw-agent-operator-test'
+      group: 'e2e-cw-agent-operator-test'
```

Comment:
Why the rename?

Reply:
If we change the group name to the cluster name it's more clear that the limiting factor for concurrency is the cluster. I'll add a comment to ensure people use it properly